model/result/parser/XmlResultParser.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA
+ */
+namespace oat\taoLtiConsumer\model\result\parser;
+
+use DOMDocument;
+use DOMXPath;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoLtiConsumer\model\result\MessageBuilder;
+use oat\taoLtiConsumer\model\result\parser\dataExtractor\DataExtractorInterface;
+use oat\taoLtiConsumer\model\result\ResultException;
+
+/**
+ * Class XmlResultParser
+ * Class to parse XML result data
+ *
+ * @author Moyon Camille
+ */
+class XmlResultParser extends ConfigurableService
+{
+    public const SERVICE_ID = 'taoLtiConsumer/xmlResultParser';
+    public const OPTION_DATA_EXTRACTORS = 'extractors';
+
+    /**
+     * Parse $xml to extract data based on configured extractors
+     *
+     * @param string $xml
+     *
+     * @return LisOutcomeRequest
+     * @throws ResultException
+     */
+    public function parse($xml)
+    {
+        $xpath = $this->load($xml);
+        $dataExtractor = $this->getDataExtractor($xpath);
+        $requestType = $dataExtractor->getRequestType();
+        $data = $dataExtractor->getData($xpath);
+
+        $this->close($xpath);
```

Comment:
It does nothing as I understand:
1. `$this->close()` unsets it's own copy of variable
2. even replacing it by `unset($xpath)` seems to be useless, because the variable will be cleared after method returns anyway